ironfish/src/wallet/wallet.ts

```diff
@@ -1002,14 +1001,59 @@ export class Wallet {
     }
   }
 
-  private buildAmountsNeeded(
+  async fundFromNoteHashes(
     raw: RawTransaction,
     options: {
+      account: Account
+      noteHashes: Buffer[]
+      fee?: bigint
+      feeRate?: bigint
+    },
+  ): Promise<void> {
+    const amountsNeeded = this.buildAmountsNeeded(raw)
+    const amountsSpent = new AssetBalances()
+
+    for (const noteHash of options.noteHashes) {
+      const decryptedNote = await options.account.getDecryptedNote(noteHash)
+      Assert.isNotUndefined(decryptedNote)
```

Comment:
Do you want to make error message here a little more explicit

Reply:
Added! 👍